lib/council-server/src/server.rs

```diff
@@ -105,71 +161,100 @@ impl Server {
             };
 
-            match request {
+            // Cache the reply channel in case we are missing dependency data and need to restart.
+            let cached_reply_channel = reply_channel.clone();
+
+            let (result, discrim) = match request {
                 Request::ValueDependencyGraph {
                     change_set_id,
                     dependency_graph,
-                } => {
+                } => (
                     register_graph_from_job(
-                        &mut complete_graph,
+                        complete_graph,
                         reply_channel,
                         change_set_id,
                         dependency_graph,
                     )
-                    .await
-                    .unwrap();
-                }
+                    .await,
+                    RequestDiscriminants::ValueDependencyGraph,
+                ),
                 Request::ProcessedValue {
                     change_set_id,
                     node_id,
-                } => {
+                } => (
                     job_processed_a_value(
                         &self.nats,
-                        &mut complete_graph,
+                        complete_graph,
                         reply_channel,
                         change_set_id,
                         node_id,
                     )
-                    .await
-                    .unwrap();
-                }
-                Request::Bye { change_set_id } => {
-                    job_is_going_away(&mut complete_graph, reply_channel, change_set_id)
-                        .await
-                        .unwrap();
-                }
+                    .await,
+                    RequestDiscriminants::ProcessedValue,
+                ),
+                Request::Bye { change_set_id } => (
+                    job_is_going_away(complete_graph, reply_channel, change_set_id).await,
+                    RequestDiscriminants::Bye,
+                ),
                 Request::ValueProcessingFailed {
                     change_set_id,
                     node_id,
-                } => {
+                } => (
                     job_failed_processing_a_value(
                         &self.nats,
-                        &mut complete_graph,
+                        complete_graph,
                         reply_channel,
                         change_set_id,
                         node_id,
                     )
-                    .await
-                    .unwrap();
+                    .await,
+                    RequestDiscriminants::ValueProcessingFailed,
+                ),
+                Request::Restart => {
+                    debug!(
+                        %management_channel, %management_reply_channel, "found restart request sent to everyone subscribing to management channel: no-op"
+                    );
+                    (Ok(()), RequestDiscriminants::Restart)
```

Comment:
Since council subscribes to `council.*`, we will see the restart message. This is a "no-op".